include/openthread/ip6.h

```diff
@@ -370,13 +370,14 @@ otError otIp6CreateSemanticallyOpaqueIid(otInstance *aInstance, otNetifAddress *
  * Allocate a new message buffer for sending an IPv6 message.
  *
  * @param[in]  aInstance             A pointer to an OpenThread instance.
- * @param[in]  aLinkSecurityEnabled  TRUE if the message should be secured at Layer 2
+ * @param[in]  aLinkSecurityEnabled  TRUE if the message should be secured at Layer 2.
+ * @param[in]  aPriority             The message priority level.
  *
  * @returns A pointer to the message buffer or NULL if no message buffers are available.
  *
  * @sa otFreeMessage
  */
-otMessage *otIp6NewMessage(otInstance *aInstance, bool aLinkSecurityEnabled);
+otMessage *otIp6NewMessage(otInstance *aInstance, bool aLinkSecurityEnabled, otMessagePriority aPriority);
```

Comment:
Rather than continuing to add on parameters, how about we define a new struct that includes both link security and priority. Then each `NewMessage()` function can take the struct pointer. If `NULL` is passed, then default values are used. Otherwise, the values in the structure are used. This makes it easier to expand the set of parameters in the future without having to change the function signature. Thoughts?

Reply:
Indeed, It is better to expand the parameters in the future if we are using a structure pointer as the argument of the `NewMessage`. I will update the code.